tools/SourceKit/cmake/modules/AddSwiftSourceKit.cmake

```diff
@@ -129,12 +129,11 @@ function(add_sourcekit_swift_runtime_link_flags target path HAS_SWIFT_MODULES)
       target_link_libraries(${target} PRIVATE "swiftCore")
 
       target_link_directories(${target} PRIVATE ${host_lib_dir})
-      if(ASKD_BOOTSTRAPPING_MODE STREQUAL "HOSTTOOLS")
-        list(APPEND RPATH_LIST "${host_lib_dir}")
-      else()
-        file(RELATIVE_PATH relative_rtlib_path "${path}" "${SWIFTLIB_DIR}/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
-        list(APPEND RPATH_LIST "$ORIGIN/${relative_rtlib_path}")
-      endif()
+
+      file(RELATIVE_PATH relative_rtlib_path "${path}" "${SWIFTLIB_DIR}/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
+      list(APPEND RPATH_LIST "$ORIGIN/${relative_rtlib_path}")
+      # NOTE: SourceKit components are NOT executed before stdlib is built.
+      # So there's no need to add RUNPATH to builder's runtime libraries.
 
     elseif(ASKD_BOOTSTRAPPING_MODE STREQUAL "BOOTSTRAPPING")
       get_bootstrapping_swift_lib_dir(bs_lib_dir "")
```

Comment:
Ah, (4) adds host. IMO I'd move that to inside the ifs here since the if there is wrong anyway (it should be specifically linux/android/openbsd). So ie.
```
if(SWIFT_SWIFT_PARSER)
    file(RELATIVE_PATH relative_hostlib_path "${path}" "${SWIFTLIB_DIR}/host")
    list(APPEND RPATH_LIST "$ORIGIN/${relative_hostlib_path}")
endif()
```

Reply:
:+1: Done